packages/aws-cdk-lib/aws-cloudfront/lib/experimental/nodejs-edge-function.ts

```
@@ -0,0 +1,64 @@
+import { Construct } from 'constructs';
+import { EdgeFunction } from './edge-function';
+import { NodejsFunctionProps } from '../../../aws-lambda-nodejs';
+import { Bundling } from '../../../aws-lambda-nodejs/lib/bundling';
+import { getRuntime, resolveBundlingConfig, validateNodejsRuntime } from '../../../aws-lambda-nodejs/lib/function-helpers';
+import { ValidationError } from '../../../core';
+
+/**
+ * Properties for a NodejsEdgeFunction
+ */
+export interface NodejsEdgeFunctionProps extends NodejsFunctionProps {
+  /**
+   * The stack ID of Lambda@Edge function.
+   *
+   * @default - `edge-lambda-stack-${region}`
+   */
+  readonly stackId?: string;
+}
+
+/**
+ * A Node.js Lambda@Edge function bundled using esbuild
+ *
+ * @resource AWS::Lambda::Function
+ */
+export class NodejsEdgeFunction extends EdgeFunction {
+  constructor(scope: Construct, id: string, props: NodejsEdgeFunctionProps = {}) {
+    validateNodejsRuntime(scope, props.runtime);
+
+    const runtime = getRuntime(scope, props);
+
+    if (props.code !== undefined) {
```

I don't think we need to handle this, we can throw an error from here.